src/client.cpp

```diff
@@ -758,8 +754,18 @@ void CClient::OnSndCrdReinitRequest ( int iSndCrdResetType )
     }
     MutexDriverReinit.unlock();
 
+    if ( !strError.isEmpty() )
+    {
+#ifndef HEADLESS
+        QMessageBox::critical ( 0, APP_NAME, strError, tr ( "Ok" ) );
+#else
+        qCritical() << qUtf8Printable ( strError );
+        exit ( 1 );
+#endif
+    }
+
     // inform GUI about the sound card device change
-    emit SoundDeviceChanged ( strError );
+    emit SoundDeviceChanged();
```

Comment:
Why not leave it in the signal?

Ideally we'd have two slots - one for GUI mode and one for headless mode.

Reply:
It's probably just because of a change in error handling with the message box class he wanted to implement. I don't think it's needed here -> revert it

Reply:
Reverted